core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -170,7 +171,16 @@ trait EndToEndAuthorizationTest extends IntegrationTestHarness with SaslSetup {
     * Tests the ability of producing and consuming with the appropriate ACLs set.
     */
   @Test
-  def testProduceConsume {
+  def testProduceConsumeViaAssign {
```

Comment:
Testing via assign and via subscribe sounds fine, but I'm not sure what behavior you're trying to test here. Could you clarify, please?